jdbc-driver/src/test/java/com/google/cloud/dataproc/jdbc/DataprocInfoTest.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import static com.google.common.truth.Truth.assertThat;
+import static io.grpc.Status.Code.NOT_FOUND;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.api.gax.grpc.GrpcStatusCode;
+import com.google.api.gax.rpc.ApiException;
+import com.google.cloud.dataproc.v1.Cluster;
+import com.google.cloud.dataproc.v1.ClusterControllerClient;
+import io.grpc.Status;
+import java.util.Collections;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+public class DataprocInfoTest {
+    private final String PROJECT_ID = "pid";
+    private final String REGION = "us-central1";
+    private final String FILTER_DUPLICATE =
+            "status.state = RUNNING AND clusterName = simple-cluster1";
+    private final String FILTER_DEFAULT = "status.state = RUNNING";
+    private final String FILTER_LONG =
+            "status.state = RUNNING AND labels.com = google AND labels.env = staging AND labels.team ="
+                    + " dataproc";
+    private final String CLUSTER_NAME_1 = "simple-cluster1";
+    private final String CLUSTER_NAME_2 = "simple-cluster2";
+    private final String NO_CLUSTER_NAME = "no-cluster";
+
+    private HiveJdbcConnectionOptions.Builder paramBuilder;
+    private DataprocInfo infoTest;
+
+    // Mock CluterControllerClient
+    ClusterControllerClient mockClusterControllerClient;
+
+    // Mock ListClustersPagedResponse
+    ClusterControllerClient.ListClustersPagedResponse mockListDefault;
+    ClusterControllerClient.ListClustersPagedResponse mockListDefaultDup;
+
+    // Mock clusters
+    Cluster mockCluster1;
+    Cluster mockCluster2;
+
+    @Before
+    public void setUp() {
+        mockClusterControllerClient = mock(ClusterControllerClient.class);
+        mockListDefault = mock(ClusterControllerClient.ListClustersPagedResponse.class);
+        mockListDefaultDup = mock(ClusterControllerClient.ListClustersPagedResponse.class);
+        mockCluster1 = mock(Cluster.class);
```

Comment:
As discussed offline, you can have mocks build and return real `Cluster` objects that contain the information you need -- such as a cluster name, labels, status.state, etc.

I learned today that rather than mocking stub classes, you can actually create a mock server instead: https://grpc.github.io/grpc-java/javadoc/io/grpc/inprocess/InProcessServerBuilder.html

Example: https://g3doc.corp.google.com/net/grpc/g3doc/java/unit_test.md?cl=head